db/kv_dense_block.go

```diff
@@ -594,7 +595,10 @@ func (d *DenseBlock) MakeLogEntry(indexEntry DenseBlockIndexEntry, entry DenseBl
 }
 
 func (d *DenseBlock) GetIndexEntry(position int64) (indexEntry DenseBlockIndexEntry) {
-	indexEntry = d.value[position : position+INDEX_ENTRY_LEN]
+	indexEntry, err := base.SafeSlice(d.value, int(position), int(position+INDEX_ENTRY_LEN))
```

Comment:
I think there are a couple more places where we do slice targeting in dense_block - it feels like we should at least switch over GetEntry at the same time.